### PR TITLE
docs: Create p022_enodebd_enhancements.md

### LINKED ISSUE #10412: eNodeBd Enhancements for Cell PnP Support
The objective of this issue is to add services to address the logistics of scale radio deployments using Magma. Specifically, plug-n-play onboarding and firmware update via TR-069.

Plug-n-play onboarding enables auto-discovery and configuration of a new radio connected to a Magma network. This will enable less sophisticated users to add radios to a Magma managed network, as well making it quicker and easier for private cellular deployment users to setup and/or scale their networks.

Firmware update via TR-069 will allow the enodebd ACS to update the firmware on radios connected to the Magma network. Automated and Remote firmware updates are an important requirement for any scale deployment.

Design doc: https://github.com/magma/magma/pull/10404/files

### PR DESCRIPTION
**Pull request by Arsenii-Oganov**
Proposal tracked in: https://github.com/magma/magma/issues/10412

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by electronjoe** [comment]
@Arsenii-Oganov thank you for the proposal! In browsing your PR doc I really appreciate the level of detail you have provided.

This looks like a proposal - we have a process for proposals to ensure the community is aware of them / try to coordinate the future of Magma.  Would you mind filing a GH Issue associated with this more detailed proposal `Design Doc` (that is what I would call this PR - which is indeed part of our process)?

See the [Magma Proposal Process](https://docs.magmacore.org/docs/proposals/p014_proposal_process).

Again, thanks for the level of detail! Heads up @hcgatewood (TSC chair, manages proposal liveliness).

**Review by electronjoe — COMMENTED** [review]
@Arsenii-Oganov thank you for the proposal! In browsing your PR doc I really appreciate the level of detail you have provided.

This looks like a proposal - we have a process for proposals to ensure the community is aware of them / try to coordinate the future of Magma.  Would you mind filing a GH Issue associated with this more detailed proposal `Design Doc` (that is what I would call this PR - which is indeed part of our process)?  After you have filed the GH Issue, maybe tag this PR description with the magma GH Issue #.

See the [Magma Proposal Process](https://docs.magmacore.org/docs/proposals/p014_proposal_process).

Again, thanks for the level of detail! Heads up @hcgatewood (TSC chair, manages proposal liveliness).

**Comment by Arsenii-Oganov** [comment]
> @Arsenii-Oganov thank you for the proposal! In browsing your PR doc I really appreciate the level of detail you have provided.
> 
> This looks like a proposal - we have a process for proposals to ensure the community is aware of them / try to coordinate the future of Magma. Would you mind filing a GH Issue associated with this more detailed proposal `Design Doc` (that is what I would call this PR - which is indeed part of our process)? After you have filed the GH Issue, maybe tag this PR description with the magma GH Issue #.
> 
> See the [Magma Proposal Process](https://docs.magmacore.org/docs/proposals/p014_proposal_process).
> 
> Again, thanks for the level of detail! Heads up @hcgatewood (TSC chair, manages proposal liveliness).

Will do shortly

**Comment by Arsenii-Oganov** [comment]
@electronjoe I've created and issue and also created commit to this proposal with adding the link to issue

**Comment by hcgatewood** [comment]
Fantastic level of detail! I went ahead and linked the proposal with this doc. I'll forward it to the relevant Magma TL and we can link y'all up 🙂

File: docs/readmes/proposals/p022_enodebd_enhancements.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+- Sercomm Englewood 
+
+### **2 Background**
+
+**2.1: Terminology**
+
+1. **TR-069** refers to an application layer technical specification of the CPE WAN Management Protocol (CWMP) both specified by the Broadband Forum. This application is used for the remote management of configuration, firmware, and metrics in network equipment
+
+2. **ACS** refers to the Automatic Configuration Server which is the TR-069 server in a server client model.
+
```
Comment (ardzoht) [inline docs/readmes/proposals/p022_enodebd_enhancements.md:31]: We should call out explicitly here the terminology and differences between magma `Managed` and `Externally managed` eNodeb devices. 

```suggestion
3. Magma **managed** eNBs refer to the radio devices that use TR-069 server communication using the enodebd service. **Externally managed** eNBs, in contrast, do not use TR-069 and are expected to be already configured to match the Cell ID and other configuration parameters so they successfully get connected to the AGW.
```

File: docs/readmes/proposals/p022_enodebd_enhancements.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+When Auto-discovery is enabled, enodebd will receive a new inform, check it’s local configuration to see if the serial number is known, if it is not, it will forward the serial number and device ID field contents to Control Proxy with the endpoint specified as Onboarding Service. 
+
+**3.1.3.2 Control Proxy Service (Modified)**
+The Control Proxy service will be modified to add Onboarding Service onboardd to the service registry. This new endpoint will be used for transporting messages between enodebd and onboardd.
+
+**3.1.3.3 Onboardd Service (New)**
+The Onboarding Service onboardd will be a new service added to support automated radio onboarding. The primary function will be to respond to enodebd new radio messages and using the API, create a new radio element and assign it to the source AGW. 
+
+In addition, when the new eNB is a CBRS CBSD, onboardd will also create a new radio element in the Domain Proxy (DP). When the radio is a Cat A indoor with a GPS derived location, the onboardd configuration will be sufficient for the DP to register and obtain a grant for the new eNB. 
+
```
Comment (ardzoht) [inline docs/readmes/proposals/p022_enodebd_enhancements.md:99]: Let's also add a description here to note that the `onboardd` service and plug and play feature will only take into account `Managed` eNBs due to the TR-069 session.

```suggestion
The onboardd service will only support magma Managed eNodebs, this is due to the requirement of establishing a TR-069 server session to further configure the radio device. Externally managed devices do not setup a TR-069 session.
```

**Comment by ardzoht** [comment]
@Arsenii-Oganov looks like the commit is not passing the DCO check job. You can refer to https://docs.magmacore.org/docs/next/contributing/contribute_workflow#guidelines to complete the sign-off of your commits.

**Comment by Arsenii-Oganov** [comment]
> @Arsenii-Oganov looks like the commit is not passing the DCO check job. You can refer to https://docs.magmacore.org/docs/next/contributing/contribute_workflow#guidelines to complete the sign-off of your commits.

@ardzoht I've signed last commit

**Review by ardzoht — APPROVED** [review]
lgtm, thanks for the updates!

**Review by hcgatewood — APPROVED** [review]
Lgtm!

**Comment by hcgatewood** [comment]
Please fix Markdown lint check, then we can force-merge!

**Comment by Arsenii-Oganov** [comment]
> Please fix Markdown lint check, then we can force-merge!

fixed. force-committed

**Comment by ardzoht** [comment]
@Arsenii-Oganov looks like markdown check is still complaining on the added Markdown file, can you take a look? https://github.com/magma/magma/runs/4349471557?check_suite_focus=true

**Review by electronjoe — APPROVED** [review]
Thank you for the detailed proposal!

**Pull request merged**